src/main/scala/ai/privado/languageEngine/python/passes/config/PythonPropertyFilePass.scala

```diff
@@ -0,0 +1,178 @@
+import java.util.Properties
+import scala.util.{Failure, Success, Try}
+import io.shiftleft.semanticcpg.language._
+import com.typesafe.config._
+
+import java.io.File
+import scala.io.Source
+
+class PythonPropertyFilePass(cpg: Cpg, projectRoot: String) extends ForkJoinParallelCpgPass[String](cpg) {
+  override def generateParts(): Array[String] = {
+    configFiles(projectRoot, Set(".ini", ".env", ".conf")).toArray
+  }
+
+  private val logger = LoggerFactory.getLogger(getClass)
+
+  override def runOnPart(builder: DiffGraphBuilder, file: String): Unit = {
+    val fileNode      = addFileNode(file, builder)
+    val propertyNodes = addPropertyNodesAndConnectToUsers(file, builder)
+    propertyNodes.foreach(builder.addEdge(_, fileNode, EdgeTypes.SOURCE_FILE))
+  }
+
+  private def addPropertyNodesAndConnectToUsers(
+    file: String,
+    builder: BatchedUpdate.DiffGraphBuilder
+  ): List[NewJavaProperty] = {
+    Try {
+      obtainKeyValuePairs(file).filter(pair => pair._1.nonEmpty && pair._2.nonEmpty)
+    } match {
+      case Success(keyValuePairs) =>
+        val propertyNodes = keyValuePairs.map(addPropertyNode(_, builder))
+
+        propertyNodes.foreach(propertyNode => {
+          connectGetEnvironLiterals(propertyNode, builder)
+          connectDBConfigMembers(propertyNode, builder)
+        })
+
+        propertyNodes
+      case Failure(exception) =>
+        logger.warn(exception.getMessage)
+        List()
+    }
+  }
+
+  private def obtainKeyValuePairs(file: String): List[(String, String)] = {
+    if (file.endsWith(".ini")) {
+      parseINIFiles(file)
+    } else {
+      getDotenvKeyValuePairs(file)
+    }
+  }
+
+  private def getDotenvKeyValuePairs(file: String): List[(String, String)] = {
+    val envProps = new Properties()
+    Source
+      .fromFile(file)
+      .getLines()
+      .filter(line => line.trim.nonEmpty && !line.startsWith("#"))
+      .foreach(line => {
+        val Array(key, value) = line.split("=", 2);
+        envProps.setProperty(key, value)
+      })
+
+    envProps.asScala
+      .map(prop => (prop._1, prop._2))
+      .toList
+  }
+
+  /** Matches the exact key of the propertyNode to its corresponding os.environ.get() calls.
+    */
+  private def matchEnvironGetCalls(propertyName: String): List[Literal] = {
+    cpg.literal
+      .codeExact("\"" + propertyName + "\"")
+      .where(_.inCall.methodFullName(".*\\(?environ\\)?\\.get"))
+      .l
+  }
+
+  /** Gets all files to be processed as property files.
+    */
+  private def configFiles(projectRoot: String, extensions: Set[String]): List[String] = {
+
+    def getListOfFiles(dir: String): List[File] = {
+      val d = new File(dir)
+      if (d.exists && d.isDirectory) {
+        d.listFiles.filter(_.isFile).toList
+      } else {
+        List[File]()
+      }
+    }
+
+    SourceFiles
+      .determine(Set(projectRoot), extensions)
+      .concat(getListOfFiles(projectRoot).map(f => f.getAbsolutePath).filter(_.matches(".*\\.env.*")))
```

Comment:
@karan-batavia Why did you removed `.filter(Utilities.isFileProcessable)` , I asked to add this to the output of getListOfFiles so that we filter invalid property files